### DISCUSSION TITLE
Has anyone used a templating system with git-crypt?

### DISCUSSION THREAD

**Discussion by qubitz** [AGWA/git-crypt#259] [Q&A] [answered]
I have configuration files (mostly yaml) that include helpful setup options and harmful secrets. I'd like to share these files publicly for the setup options, but git-crypt encrypts an *entire* file rather than just a particular section of a file.

[Handlebars](https://handlebarsjs.com/) or [Mustache](https://mustache.github.io/) seem like good templating solutions to split secrets into there own files to be encrypted. Ideally the merging and splitting of these files would be done transparently. Has anyone else out there already made a working solution? Is there an easy integration I'm missing? How could I extend git-crypt to include this functionality?

**Comment by alerque** [comment, marked as answer]
Some/many YAML based configuration files have support for some sort of import command or can load multiple YAML files. Look into that first because it makes it easy to split secrets in one file and everything else in others and just load them all.

Beyond that, almost any templating or macro language could handle this. The correct answer for your use case probably depends more on your build / deploy system than anything else. Even `m4` could be used, but hes of course handlebars or mustache or tera or any of a host of others would work. Use what's easy to run and have on hand in your build/deploy system.
